pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -1011,4 +1012,33 @@ public void testStatementDescribe() throws SQLException {
     pstmt.close();
   }
 
+  @Test
+  public void testSetTimeAndTimestamp() throws SQLException {
+    PreparedStatement ps = con.prepareStatement("SELECT ?::timestamp");
+    try {
+      Timestamp ts = new Timestamp(1474997614836L);
+      for (int i = 0; i < 3; ++i) {
+        ResultSet rs;
+        ps.setNull(1, Types.DATE);
+        rs = ps.executeQuery();
+        try {
+          assertTrue(rs.next());
+          assertNull(rs.getObject(1));
```

Comment:
I read the rules for the tests writing. But not figured, what I'm doing wrong.
Also, there is (into pgjdbc test classes) a lot of code that uses assertTrue for rs.next() and assertEquals without a message.